File: backend/components/metiers/administration_publique/utils/validators/core/validators.test.js
// validators.test.js – Tests unitaires JS pour validators Threed
const validators = require('./validators');
describe('Validators Threed', () => {
  it('valide un modèle 3D correct', () => {
    expect(validators.isValidModel({ nom: 'Test', statut: 'actif' })).toBe(true);
  });
  it('rejette un modèle 3D invalide', () => {
    expect(validators.isValidModel({})).toBe(false);
  });
});

File: backend/components/metiers/administration_publique/utils/validators/core/validators.test.py
"""
validators.test.py – Tests unitaires Python pour validators Threed
"""
from .validators import validate_threed

def test_valid_threed():
    data = {'nom': 'Test', 'statut': 'actif'}
    assert validate_threed(data)

def test_invalid_threed():
    try:
        validate_threed({'nom': '', 'statut': 'foo'})
    except ValueError:
        assert True
    else:
        assert False
